docs/security-best-practices.md

```diff
@@ -1,324 +1,82 @@
+> For more information, see the
+> [configopaque](https://pkg.go.dev/go.opentelemetry.io/collector/config/configopaque)
 > documentation.
 
-When defining Go structs for configuration data that may contain sensitive information, use the `configopaque` package to define fields with the `configopaque.String` type. This ensures that the data is masked when serialized to prevent accidental exposure.
-
-> For more information, see the [configopaque](https://pkg.go.dev/go.opentelemetry.io/collector/config/configopaque) documentation.
-
-Component developers MUST get configuration information from the Collector's
-configuration file. Component developers SHOULD leverage [configuration helper
-functions](https://github.com/open-telemetry/opentelemetry-collector/tree/main/config).
-
-More information about configuration is provided in the following sections.
-
 ## Permissions
 
 The Collector supports running as a custom user and SHOULD NOT be run as a
 root/admin user. For the majority of use-cases, the Collector SHOULD NOT require
-privileged access to function. Some components MAY require privileged access
-and care should be taken before enabling these components. Collector components
-MAY require external permissions including network access or RBAC.
+privileged access to function. Some components MAY require privileged access or
+external permissions, including network access or RBAC.
 
 Component developers SHOULD minimize privileged access requirements and MUST
 document what requires privileged access and why.
 
-More information about permissions is provided in the following sections.
-
 ## Receivers and Exporters
 
 Receivers and Exporters can be either push or pull-based. In either case, the
 connection established SHOULD be over a secure and authenticated channel.
-Unused receivers and exporters SHOULD be disabled to minimize the attack vector
-of the Collector.
-
-Receivers and Exporters MAY expose buffer, queue, payload, and/or worker
-settings via configuration parameters. If these settings are available,
-end-users should proceed with caution before modifying the default values.
-Improperly setting these values may expose the Collector to additional attack
-vectors including resource exhaustion.
-
-> It is possible that a receiver MAY require the Collector run in a privileged
-> mode in order to operate, which could be a security concern, but today this
-> is not the case.
 
-Component developers MUST default to encrypted connections (via the `insecure:
-false` configuration setting) and SHOULD leverage
+Component developers MUST default to encrypted connections (using the
+`insecure: false` configuration setting) and SHOULD leverage
 [gRPC](https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/configgrpc)
 and
 [http](https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp)
 helper functions.
 
-### Safeguards against denial of service attacks
```

Comment:
There are a lot of links to this section: https://github.com/search?q=-is%3Afork+https%3A%2F%2Fgithub.com%2Fopen-telemetry%2Fopentelemetry-collector%2Fblob%2Fmain%2Fdocs%2Fsecurity-best-practices.md%23safeguards-against-denial-of-service-attacks&type=code

To have a more graceful transition, I would suggest we keep the heading and just have the section say "see [link on opentelemetry.io] to learn how to safeguard against denial of service attacks".